cli/azd/pkg/experimentation/assignment.go

```diff
@@ -0,0 +1,186 @@
+package experimentation
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/azure/azure-dev/cli/azd/internal/tracing/resource"
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+)
+
+// cCacheFileNamePattern is the pattern used to generate the cache file name. It is formatted
+// using the machine ID.
+const cCacheFileNamePattern = "mid-%s.cache"
+
+// MachineIdParameterName is the name of the parameter used to identify the machine ID in the assignment
+// request.
+const MachineIdParameterName = "machineid"
+
+// AssignmentsManager manages interaction with the Assignments service, caching the results for 24 hours.
+type AssignmentsManager struct {
+	cacheRoot string
+	client    *tasClient
+}
+
+// NewAssignmentsManager creates a new AssignmentManager, which will communicate with the TAS service. The
+// AssignmentManager caches the assignment information for 24 hours.
+
+func NewAssignmentsManager(endpoint string, transport policy.Transporter) (*AssignmentsManager, error) {
+	configRoot, err := config.GetUserConfigDir()
+	if err != nil {
+		return nil, err
+	}
+
+	cacheRoot := filepath.Join(configRoot, "experiments")
+	if err := os.MkdirAll(cacheRoot, osutil.PermissionDirectory); err != nil {
+		return nil, err
+	}
+
+	client, err := newTasClient(endpoint, &policy.ClientOptions{
+		Transport: transport,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &AssignmentsManager{
+		cacheRoot: cacheRoot,
+		client:    client,
+	}, nil
+}
+
+// Assignment is a subset of the information returned by the TAS service.
+type Assignment struct {
+	Features          []string
+	Flights           map[string]string
+	Configs           []AssignmentConfig
+	ParameterGroups   []string
+	AssignmentContext string
+}
+
+// AssignmentConfig is information about a specific config in an assignment.
+type AssignmentConfig struct {
+	ID         string
+	Parameters map[string]interface{}
+}
+
+// Assignment gets a the assignment information for this given machine.
+func (am *AssignmentsManager) Assignment(ctx context.Context) (*Assignment, error) {
+	machineId := resource.MachineId()
+
+	cachedAssignment, err := am.readResponseFromCache(machineId)
+	if err != nil {
+		log.Printf("could not read assignment from cache: %v", err)
+	}
+
+	if cachedAssignment == nil {
+		assignment, err := am.client.GetVariantAssignments(ctx, &variantAssignmentRequest{
+			Parameters: map[string]string{
+				MachineIdParameterName: resource.MachineId(),
```

Comment:
@weikanglim, @savannahostrowski  Are there other parameters that we want to include as part of this request or is just machine ID sufficient?

Reply:
Looks good to me. Prefer keeping it simple for now.